### PR TITLE
Add German Building Database

### PR DESCRIPTION
**Pull request by ddceruti**
German building stock database contribution by @Mar-Ges, @ululele and @ddceruti for CEA. Tested on CEA v3.35.2, based on the CH database. However, it does not include the "_NEW.xslx" files. This causes a known bug when viewing or editing the database in the Dashboard, which can be quick fixed (Issue #3432).

Details on assumptions, sources and method are located in the `documentation.md` file. Additional information is in the `README.md` file. 

This database is detailed and validated in a conference paper currently under review. The citation is included in both .md files and will be updated as soon as possible. 

Tagging @ShiZhongming to notify him that the pull request has been opened.

### CODE REVIEW

**Comment by ShiZhongming** [comment]
Dear @c0nb4.

I am tagging you here as a reviewer for thie PR. Thank you for volunteering!

Best, Zhongming

File: cea/databases/DE/archetypes/use_types/D13_RESTAURANT.csv
Comment (ShiZhongming) [inline cea/databases/DE/archetypes/use_types/D13_RESTAURANT.csv:1]: Another naming_convention question: what does D13 stand for?

File: cea/databases/DE/archetypes/use_types/D_MFH.csv
Comment (ShiZhongming) [inline cea/databases/DE/archetypes/use_types/D_MFH.csv:1]: What is the difference between D_MFH and the original MULTI_RES?
Not sure I missed in the documentation about this.

Question - Do you think it makes sense that we change all the german works to English and reference them in the reference.txt in German? I assume not the people modelling German context understand German...

File: cea/databases/DE/assemblies/ENVELOPE.xlsx
Comment (ShiZhongming) [inline cea/databases/DE/assemblies/ENVELOPE.xlsx:1]: Another naming convention suggestion: Possible to change WIN_SFH_A to WINDOW_SFH_A? Thit is just to keep the naming formalities in CEA Database...

**Comment by ddceruti** [comment]
Hi Zhongming, 
thanks a lot for your suggestions! 
Of course we will change the naming to fit CEA's standards as much as possible. I will implement all the comments directly and push a new version of the database. Lastly, we will discuss the naming conventions of the use_types internally and get back to you with a new version.

File: cea/databases/DE/archetypes/CONSTRUCTION_STANDARD.xlsx
[the remark this answers is not in the document]
Comment (ddceruti) [reply cea/databases/DE/archetypes/CONSTRUCTION_STANDARD.xlsx:1]: I changed the NWG_BMVBS_1 to _A for consistency like you proposed. 
NWG_G1_A stands for a general category which includes several subtypes. In this case, for example, G1 is a broad category for categories NWG_1 to NWG_7. I updated the documentation.md clarifying this. We can also look for another naming convention if it helps.

**Comment by ddceruti** [comment]
Hi, we updated several files to adapt to the naming conventions, including the system_references.txt for the use_types by @Mar-Ges and updated some documentation.

File: cea/databases/DE/archetypes/use_types/D_MFH.csv
Comment (ShiZhongming) [reply cea/databases/DE/archetypes/use_types/D_MFH.csv:1]: @ddceruti I think the format of a central documentation is good. Maybe we should do the same for the CH and SG database. I will bring this up in the CEA Monthly Assembly next week for discussion.

**Comment by c0nb4** [comment]
Thanks for reaching out for review. 
 
The following points are general remarks. 
 
- TABULA lack's data for Apartment Blocks after 1979. Hence, data is lacking in the typology. I've seen papers just copying the MFH Typology, e.g. https://www.researchgate.net/publication/378038367_A_modular_framework_for_a_dynamic_residential_building_stock_model_with_energy_retrofit_forecasts 
 
Comments for the implementation:
 
Archetypes: 
 
  - Description of IWU Typology for Non-Residential is different from description of Residential (2016 - ...)  for 2010 / 2016 - onwards. 
  - Typo in "Non-residentail" in column description 127-130 in the standard definition. 
  - GDR buildings are currently not implemented. 
  - Abbreviation NWG is short for Nichtwohngebäude (Non-Residential), further abbreviations are English. 
  
  Use-Types ( compared with: https://linkinghub.elsevier.com/retrieve/pii/S2210670723003050  and 18599-10) 
 
- Typo in metadata for  D1-D13, in "Din 18599" 
- Schedule for D2-Buero, according to DIN18599-10 the usage time is from 7-18. In the file, it is from 8-18.  What is the reason for this? Other schedules als derivate. 
- Set points seem to be mixed up for D4, D8 (both), D6, D4 and D13. 
- Set point for D4 should be 21 degrees, according to the previous paper.

**Comment by ShiZhongming** [comment]
Hi @all 
Thanks for the updates.
I just reviewed the second version of the DE Database and coming back with the following comments.

1. Typo: `kindergarten` instead of `kindergarden`
2. ENVELOPE.xlsx - Tab Window: column name should be `GHG_win_kgCO2m2` instead of `GHG_WINDOW_kgCO2m2`
3. ENVELOPE.xlsx - Tabs Window, ROOF, WALL and FLOOR: missing columns `Service_Life_win`, `Service_Life_roof`, `Service_Life_wall`, and `GService_Life_floor`, respectively.

During the CEA Monthly Assembly next week - we will discuss the following:
1. DOUMENTATION.MD or reference.txt
2. Plans to intergrate _new.xlsx files.

**Comment by ShiZhongming** [comment]
> Thanks for reaching out for review.
> 
> The following points are general remarks.
> 
>     * TABULA lack's data for Apartment Blocks after 1979. Hence, data is lacking in the typology. I've seen papers just copying the MFH Typology, e.g. https://www.researchgate.net/publication/378038367_A_modular_framework_for_a_dynamic_residential_building_stock_model_with_energy_retrofit_forecasts
> 
> 
> Comments for the implementation:
> 
> Archetypes:
> 
>     * Description of IWU Typology for Non-Residential is different from description of Residential (2016 - ...)  for 2010 / 2016 - onwards.
> 
>     * Typo in "Non-residentail" in column description 127-130 in the standard definition.
> 
>     * GDR buildings are currently not implemented.
> 
>     * Abbreviation NWG is short for Nichtwohngebäude (Non-Residential), further abbreviations are English.
> 
> 
> Use-Types ( compared with: https://linkinghub.elsevier.com/retrieve/pii/S2210670723003050 and 18599-10)
> 
>     * Typo in metadata for  D1-D13, in "Din 18599"
> 
>     * Schedule for D2-Buero, according to DIN18599-10 the usage time is from 7-18. In the file, it is from 8-18.  What is the reason for this? Other schedules also differentiate from usage time in 18599.
> 
>     * Set points seem to be mixed up for D4, D8 (both), D6, D4 and D13.
> 
>     * Set point for D4 should be 21 degrees, according to the previous paper.

Thank you for reviewing the database as an expert in the German context. @ddceruti Could you take a look at this?

**Comment by ddceruti** [comment]
Thanks for your great feedback and detailed check of the database @c0nb4! I will try to clarify everything and think about the proposal of simply assigning MFH to the HR buildings like the publication did. We updated the use_types, but your feedback will still be implemented. I will also change the naming of NWG to the english version.
@ShiZhongming I will correct the ENVELOPE.xslx errors and add the columns which are missing - I cloned an old version of the database which did not include them yet! I will correct the typo of kindergarten.
I will be finishing up the work after the weekend, so expect an update on monday from my side.

**Comment by c0nb4** [comment]
I'm currently running a test simulation and try to continuously update feedback. Anyhow, thank you so much for this database. Really great work and I've been waiting for this since I've learned about CEA! 

In the supply file:
-  the electricity sheet, the last row, seems to have a different format. 
- CAPEX for district heating is taken from CEA Switzerland. Good source in germany might be nPro: https://www.npro.energy/main/en/5gdhc-networks/price-cost-data

**Comment by ddceruti** [comment]
Thanks again for the feedback. I added eastern germany archetypes and updated some HVAC assembly informations.

> In the supply file:
> 
>     * the electricity sheet, the last row, seems to have a different format.

I updated the format.

>     * CAPEX for district heating is taken from CEA Switzerland. Good source in germany might be nPro: https://www.npro.energy/main/en/5gdhc-networks/price-cost-data

I updated some capex and efficiencies for the DISTRICT technologies with the danish energy agency reference, which is the most complete I know of... npro has some information but is missing efficiency and lifetime values. Here is the link: https://ens.dk/en/our-services/projections-and-models/technology-data/technology-data-generation-electricity-and

**Comment by c0nb4** [comment]
Hey everyone,

I'm not sure on the current time frame, so I will update as long as this pull request has not been merged. 

I do have two more questions: 

- Why is the type_cs mostly HVAC_Cooling_AS3? Shouldn't it be "none" so HVAC_Cooling_AS0? I'm refering to file: "Construction_Standard" page "HVAC_Assemblies"?
- Where is Ea_WM2 "Peak specific electrical load due to computers and devices [W/m2]" defined?

**Comment by ddceruti** [comment]
Hi,

thanks again for your feedback @c0nb4 ! We will re check the schedules with @mar-ges

I updated the database with the service life, changed the cooling to AS0 (none) and the typo in GHG emissions of the windows. @ShiZhongming , what are we missing? What else do you need?

**Comment by ShiZhongming** [comment]
Hi @ddceruti 

Sorry that I forgot to update you that we did not finalise this PR as @MatNif and I are working to merge the _NEW.xlsx files.
We thought we it would be better that we fix the database and release the DE database both together for the upcoming release.

For the DE database itself, I just tested it a few times myself and I am coming back with two comments.
1. why are some cells highlighted in yellow as follows?
Screenshot 2024-04-10 at 18 27 27
2. The following cells in Columns E and Rows 149-152 seem to be a typo.
Screenshot 2024-04-10 at 18 26 11

After addressing these two comments and the newer ones from @c0nb4, I would say the DE Database is ready to be merged. We will include it into the upcoming release.

Thank you.
Best, Zhongming

**Comment by ShiZhongming** [comment]
Hi @ddceruti 

Did you have a chance to take a look at the comments above?
We are ready to merge this PR for the upcoming release.

Best, Zhongming

**Comment by ddceruti** [comment]
Hi all,
I updated the database with your comments. All ready from my part.

**Comment by ShiZhongming** [comment]
The DE Database is now merged.

Thank you @ddceruti @Mar-Ges @ululele.
Speical thanks to @c0nb4 for the reviewing work.

**Pull request merged**